gcc/rust/rust-linemap.h

```diff
@@ -33,38 +33,73 @@
 class Linemap
 {
 public:
-  Linemap ()
+  Linemap () : in_file_ (false)
   {
     // Only one instance of Linemap is allowed to exist.
     rust_assert (Linemap::instance_ == NULL);
     Linemap::instance_ = this;
   }
 
-  virtual ~Linemap () { Linemap::instance_ = NULL; }
+  ~Linemap () { Linemap::instance_ = NULL; }
 
   // Subsequent Location values will come from the file named
   // FILE_NAME, starting at LINE_BEGIN.  Normally LINE_BEGIN will be
   // 0, but it will be non-zero if the Rust source has a //line comment.
-  virtual void start_file (const char *file_name, unsigned int line_begin) = 0;
+  void start_file (const char *file_name, unsigned int line_begin)
```

Comment:
It would be better if the implementation for these methods was kept in `rust-linemap.cc`